packages/okta-adapter/src/filters/group_schema_field_removal.ts

```diff
@@ -0,0 +1,74 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { Change, InstanceElement, ModificationChange, getChangeData, isInstanceChange, isModificationChange } from '@salto-io/adapter-api'
+import { resolvePath } from '@salto-io/adapter-utils'
+import { FilterCreator } from '../filter'
+import { GROUP_SCHEMA_TYPE_NAME } from '../constants'
+
+const CUSTOM_ADDITONAL_PROPERTIES_PATH = ['definitions', 'custom', 'properties', 'additionalProperties']
+
+const addNullToRemovedProperties = (change: ModificationChange<InstanceElement>): void => {
+  const { before, after } = change.data
+  const customPropertiesPath = before.elemID.createNestedID(...CUSTOM_ADDITONAL_PROPERTIES_PATH)
+  const beforeCustomProperties = resolvePath(before, customPropertiesPath)
+  const afterCustomProperties = resolvePath(after, customPropertiesPath)
+  const afterCustomPropertiesKeys = Object.keys(afterCustomProperties)
+  const PropertiesToRemove = Object.keys(beforeCustomProperties)
+    .filter(key => !(afterCustomPropertiesKeys).includes(key))
```

Comment:
nit: the parentheses here seem off (did you mean `!afterCustomPropertiesKeys.includes(key)`?)
also - if there are a lot of custom properties this can be inefficient. I think we should either use a set, or check if 
```suggestion
    .filter(fieldName => Object.prototype.hasOwnProperty.call(afterCustomProperties, fieldName))
```
(probably not critical for most cases here)

Reply:
Changed to Set